### PR TITLE
Make High DPI Monitors work properly with Hazel!

### PR DESCRIPTION
**Pull request by standtall007**
This is the changes from previous merges plus #335.

### CODE REVIEW

**Comment by LovelySanta** [comment]
@TheCherno mentioned that he would look at this, but he didn't do it yet. He said it looked good, but he also mentioned he's not a fan of high dpi monitors :D I'll leave it here for now...

File: Hazel/src/Hazel/ImGui/ImGuiLayer.cpp [hunk 1]
```diff
@@ -33,15 +33,18 @@ namespace Hazel {
 		//io.ConfigFlags |= ImGuiConfigFlags_ViewportsNoTaskBarIcons;
 		//io.ConfigFlags |= ImGuiConfigFlags_ViewportsNoMerge;
 
-		io.Fonts->AddFontFromFileTTF("assets/fonts/opensans/OpenSans-Bold.ttf", 18.0f);
-		io.FontDefault = io.Fonts->AddFontFromFileTTF("assets/fonts/opensans/OpenSans-Regular.ttf", 18.0f);
+		io.Fonts->AddFontFromFileTTF("assets/fonts/opensans/OpenSans-Bold.ttf", Window::s_HighDPIScaleFactor * 18.0f);
+		io.FontDefault = io.Fonts->AddFontFromFileTTF("assets/fonts/opensans/OpenSans-Regular.ttf", Window::s_HighDPIScaleFactor * 18.0f);
```
[the remark this answers is not in the document]
Comment (LovelySanta) [reply Hazel/src/Hazel/ImGui/ImGuiLayer.cpp:37]: I think a single variable named s_UIScaleFactor would be more appropriate, it should scale everything equivalently...

[the remark this answers is not in the document]
Comment (standtall007) [reply Hazel/src/Hazel/ImGui/ImGuiLayer.cpp:37]: @LovelySanta Is that something i should change now or...?

[the remark this answers is not in the document]
Comment (LovelySanta) [reply Hazel/src/Hazel/ImGui/ImGuiLayer.cpp:37]: I mean... sure, if you want to, I just mentioned it as a comment to what Dreaner said

[the remark this answers is not in the document]
Comment (standtall007) [reply Hazel/src/Hazel/ImGui/ImGuiLayer.cpp:37]: Oh... i thought you meant that it should be implemented... Sorry for misunderstanding you 😞 If it is required then i would rather wait if you don't mind 😃

**Comment by DaveAxiom** [comment]
Font points are vertical measurements for fonts. Why does this patch take the horizontal screen scaling factor?

**Comment by standtall007** [comment]
> 
> 
> Font points are vertical measurements for fonts. Why does this patch take the horizontal screen scaling factor?

That's an oversight on my behalf, i'll fix it soon.

**Comment by standtall007** [comment]
This will hopefully be the last change...
I have changed the windowswindow.cpp to use yscale instead of xscale, but this is getting pretty tedious because i don't have a github app, so i do everything via commandprompt which is anything but ideal!

But look at this pull request instead: #372.

**Pull request closed without merging**